File: JAVA/Login/EmailOtpStrategy.java
public class EmailOtpStrategy implements LoginStrategy {
    @Override
    public boolean login() {
        // Implement email/OTP login logic here
        // Return true if the login is successful, replace with actual logic
        return true;
    }

    @Override
    public String getName() {
        return "Email/OTP";
    }
}

File: C++/login/emailotpstrategy.cpp
#include "emailotpstrategy.hpp"
#include "../user/normaluser.hpp"

User* EmailOtpStrategy::login() {
    // Implement email/OTP login logic here
    return new NormalUser("normal user"); // Replace with actual logic
}

string EmailOtpStrategy::getName() {
    return "Email/OTP";
}
